app/lib/shared/redis_cache.dart

```diff
@@ -223,3 +225,57 @@ Future _withInmemoryCache(FutureOr Function() fn) async {
   _registerCache(CachePatterns._(Cache(Cache.inMemoryCacheProvider(4096))));
   return await fn();
 }
+
+typedef CacheProviderFn<T> = Future<CacheProvider<T>> Function();
+
+class _UpdatingCacheProvider<T> implements CacheProvider<T> {
+  final CacheProviderFn<T> _fn;
+  CacheProvider<T> _delegate;
+  Timer _timer;
+  bool _isClosed = false;
+
+  _UpdatingCacheProvider._(this._fn, this._delegate) {
+    _timer = Timer.periodic(Duration(hours: 1), (_) async {
```

Comment:
nit, maybe make `Duration` a parameters... and then add a little jitter, like:
`Duration(minutes: 55 + Random.secure().nextInt(10));`